File: frontend/src/types/specDefinition.ts
export type SpecDataType = 'text' | 'number' | 'select';

export interface SpecDefinitionOption {
  id: number;
  optionValue: string;
  displayOrder: number;
}

export interface SpecDefinitionOptionCreateRequest {
  optionValue: string;
  displayOrder?: number;
}

export interface SpecDefinition {
  id: number;
  categoryId: number;
  keyCode: string;
  displayName: string;
  dataType: SpecDataType;
  unit?: string;
  displayOrder: number;
  isRequired: boolean;
  isActive: boolean;
  options?: SpecDefinitionOption[];
}

export interface SpecDefinitionCreateRequest {
  keyCode: string;
  displayName: string;
  dataType: SpecDataType;
  unit?: string;
  displayOrder?: number;
  isRequired?: boolean;
}

export interface SpecDefinitionUpdateRequest {
  keyCode: string;
  displayName: string;
  dataType: SpecDataType;
  unit?: string;
  displayOrder?: number;
  isRequired?: boolean;
  isActive?: boolean;
}

export interface ProductSpecification {
  id: number;
  productId: number;
  specDefinitionId: number;
  displayName: string;
  keyCode: string;
  dataType: SpecDataType;
  unit?: string;
  value?: string;
  valueNumber?: number;
  optionId?: number;
  optionValue?: string;
}

export interface ProductSpecificationCreateRequest {
  specDefinitionId: number;
  value?: string;
  valueNumber?: number;
  optionId?: number;
}

export interface ProductSpecificationUpdateRequest {
  specDefinitionId: number;
  value?: string;
  valueNumber?: number;
  optionId?: number;
}

File: backend/src/main/java/com/company/electricfanshop/entity/product/SpecDefinition.java
package com.company.electricfanshop.entity.product;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Table(name = "spec_definition", uniqueConstraints = {
        @UniqueConstraint(columnNames = {"category_id", "key_code"})
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SpecDefinition {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    private String keyCode;

    private String displayName;

    @Builder.Default
    private String dataType = "text";

    private String unit;

    @Builder.Default
    private Integer displayOrder = 0;

    @Builder.Default
    private Boolean isRequired = false;

    @Builder.Default
    private Boolean isActive = true;

    @OneToMany(mappedBy = "specDefinition", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("displayOrder ASC")
    private List<SpecDefinitionOption> options;
}
